### PR TITLE
 ops: tpetra-block: revise rank-1 update

### PR DESCRIPTION
**Pull request by mzuzek**
Contents:
- [x] revised overload constraints 
- [x] added extent check
- [x] added explicit type casting for coefficients
- [x] added NaN injection unit tests (impl is fine - no fix needed)

refs #448

### CODE REVIEW

File: include/pressio/ops/tpetra_block/ops_rank1_update.hpp [hunk 1]
```diff
@@ -145,13 +204,22 @@ update(T & v,         const a_Type &a,
...
+  sc_t a_{a};
+  sc_t b_{b};
+  sc_t c_{c};
+  sc_t d_{d};
+  sc_t e_{e};
+  v.update(b_, v1, a_); // v = a*v + b*v1
+  v.update(c_, v2, one); // add c*v2
+  v.update(d_, v3, one); // add d*v3
+  v.update(e_, v4, one); // add e*v4
 }
```
Comment (mzuzek) [inline include/pressio/ops/tpetra_block/ops_rank1_update.hpp:225]: @fnrizzi should we keep the commented code below ?

Comment (fnrizzi) [reply include/pressio/ops/tpetra_block/ops_rank1_update.hpp:225]: please remove it thanks!

File: tests/functional_small/ops/ops_tpetra_block_vector.cc [hunk 2]
```diff
@@ -276,8 +279,77 @@ TEST_F(tpetraBlockVectorGlobSize15BlockSize5Fixture, vector_update4_b)
...
 
+TEST_F(ops_tpetra_block, vector_update_nan1)
+{
+    auto v = pressio::ops::clone(*myVector_);
+    auto v2_h = v.getVectorView().getLocalViewHost(Tpetra::Access::ReadOnly);
+    auto v_h = Kokkos::subview(v2_h, Kokkos::ALL, 0);
+    pressio::ops::fill(v, 1.);
+    auto a = pressio::ops::clone(*myVector_);
+    pressio::ops::fill(a, 1.);
+    auto nan = pressio::ops::clone(*myVector_);
```
Comment (fnrizzi) [inline tests/functional_small/ops/ops_tpetra_block_vector.cc:290]: same comment for renaming as in other PRs

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see comment

File: tests/functional_small/ops/ops_tpetra_block_vector.cc (hunk 2 above)
Comment (mzuzek) [reply tests/functional_small/ops/ops_tpetra_block_vector.cc]: fixed on 9a96fb625f9c2ade6d37552b9d8f4b0575735265

File: include/pressio/ops/tpetra_block/ops_rank1_update.hpp (hunk 1 above)
Comment (mzuzek) [reply include/pressio/ops/tpetra_block/ops_rank1_update.hpp]: ok, thanks! fixed on 1403c478ca7a5b41af7637432de39e86a06ced1a

**Review by fnrizzi — APPROVED** [review]

**Comment by mzuzek** [comment]
thanks @mzuzek for your review and comments!

**Pull request merged**